gateway/reverse_proxy.go

```diff
@@ -538,21 +538,39 @@ func (p *ReverseProxy) ServeHTTPForCache(rw http.ResponseWriter, req *http.Reque
 	return resp
 }
 
+// This returns the expected 30 second default timeout in case
+// the ProxyDefaultTimeout value is undefined.
+func proxyDefaultTimeout(spec *APISpec) float64 {
+	if spec.GlobalConfig.ProxyDefaultTimeout > 0 {
+		return spec.GlobalConfig.ProxyDefaultTimeout
+	}
+	return 30
+}
+
+// CheckHardTimeoutEnforced checks APISpec versions for a fine grained timeout
+// value. The value is defined in seconds, but we're using float64 to enable
+// sub-second durations for tests. Changing to int would break that behaviour.
 func (p *ReverseProxy) CheckHardTimeoutEnforced(spec *APISpec, req *http.Request) (bool, float64) {
+	return checkHardTimeoutEnforced(spec, req)
+}
+
+// Returns if the request should have a hard timeout enforced and the timeout value configured.
+func checkHardTimeoutEnforced(spec *APISpec, req *http.Request) (bool, float64) {
```

Comment:
why another function?

Reply:
because it does not need *ReverseProxy

Reply:
Then let’s remove it, why would we duplicate?

Reply:
It's removing an exposed function, can we guarantee the removal is non-breaking?

Reply:
Reverted to single func due to the logger that's carried over *ReverseProxy.